### PR TITLE
Add PostgreSQL configuration resource and test

### PR DESCRIPTION
**Pull request by linuxluigi**
This pull request adds a new PostgreSQL configuration resource and a corresponding test. The resource allows users to configure various PostgreSQL settings for a DigitalOcean database cluster, similar like to MySQL.

I tried to implement all options from [godo](https://github.com/digitalocean/terraform-provider-digitalocean/blob/629bb3b8b4717cc2d6e96acf02207009d22790d7/vendor/github.com/digitalocean/godo/databases.go#L512). For the test I'm not sure if they are working correct, I guess here it will need some more work.

### CODE REVIEW

**Comment by danaelhe** [comment]
Taking a look now- thank you so much!

Also, running `make terrafmt` and pushing up the changes will fix that CI test

File: digitalocean/database/resource_database_postgres_config_test.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	"fmt"
+	"testing"
+
+	"github.com/digitalocean/terraform-provider-digitalocean/digitalocean/acceptance"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDigitalOceanDatabasePostgreSQLConfig_Basic(t *testing.T) {
+	name := acceptance.RandomTestName()
+	dbConfig := fmt.Sprintf(testAccCheckDigitalOceanDatabaseClusterPostgreSQL, name, "8.5")
```
Comment (danaelhe) [inline digitalocean/database/resource_database_postgres_config_test.go:13]: Test fail here with a `invalid cluster engine version`, need a valid version like`15`
```suggestion
	dbConfig := fmt.Sprintf(testAccCheckDigitalOceanDatabaseClusterPostgreSQL, name, "15")
```

File: digitalocean/database/resource_database_postgres_config.go [hunk 2]
```diff
@@ -0,0 +1,626 @@
...
+
+	"github.com/digitalocean/godo"
+	"github.com/digitalocean/terraform-provider-digitalocean/digitalocean/config"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceDigitalOceanDatabasePostgreSQLConfig() *schema.Resource {
+	return &schema.Resource{
```
Comment (danaelhe) [inline digitalocean/database/resource_database_postgres_config.go:16]:
```suggestion
	return &schema.Resource{
		CreateContext: resourceDigitalOceanDatabasePostgreSQLConfigCreate,
		ReadContext:   resourceDigitalOceanDatabasePostgreSQLConfigRead,
		UpdateContext: resourceDigitalOceanDatabasePostgreSQLConfigUpdate,
		DeleteContext: resourceDigitalOceanDatabasePostgreSQLConfigDelete,
		Importer: &schema.ResourceImporter{
			State: resourceDigitalOceanDatabasePostgreSQLConfigImport,
		},
```

Need to reference the CRUD functions here

**Comment by linuxluigi** [comment]
Thanks for reviewing. I applied your suggestions.

**Comment by danaelhe** [comment]
Thanks! Still working through the review- looks great though!!

**Comment by danaelhe** [comment]
I had some time today so I pushed up some changes as well. Mostly minor things, like adding `Computed: true` to schema variables where necessary and adding some helper functions. 

Thank you so much for your contribution! We should be able to get this merged and a new release cut today!

**Review by danaelhe — APPROVED** [review]

**Comment by linuxluigi** [comment]
Thank you a lot @danaelhe 💃

**Pull request merged**